### ISSUE TITLE
Could not install ipykernel in devcontainer

### ISSUE THREAD

**Issue by mjbvz** [microsoft/vscode-jupyter#7737]
Testing #7680

1. On MacOS
1. Open the `jupyter-datascience-notebooks` dev container from https://github.com/Microsoft/vscode-dev-containers
2. Follow instructions in test plan item up to running the cell
3. Click prompt to install `ipykernel`

**Bug**
I get a error notification :

```
Could not install ipykernel. If pip is not available, please use the package manager of your choice to manually install this library into your Python environment.
```

Potentially relevant other logs from dev tools:

![Screen Shot 2021-09-28 at 4 16 08 PM](https://user-images.githubusercontent.com/12821956/135178137-be1f0495-ea27-42af-82de-7ab7a5b7529e.png)

I was using the following kernel:

![Screen Shot 2021-09-28 at 4 17 24 PM](https://user-images.githubusercontent.com/12821956/135178255-b6ac9f82-f3d8-45e3-987d-350d0a952131.png)

**Comment by mjbvz** [comment]
Hm looks like I'm not using the kernel from virtualevn I just created, however it doesn't show up in the list of kernels:

![Screen Shot 2021-09-28 at 4 20 05 PM](https://user-images.githubusercontent.com/12821956/135178429-f2cfffc8-27e5-40cc-8af6-40f1e2ba6440.png)

**Comment by rchiodo** [comment]
@mjbvz Two questions:

- Did you have the latest python insiders installed?
- Did you create the virtual env before starting VS code?

The pip error means you have to get pip yourself. Not sure why we don't just run 'ensurepip' but here's the directions:
https://pip.pypa.io/en/stable/installation/

**Comment by mjbvz** [comment]
1. Yes I believe so. Here's the extra information:

- Python ext version: v2021.10.1283853053-dev
- Jupyter ext version: v2021.9.1001281857
- Host OS: MacOS 11.6
- Remote OS: Linux e8ce3374e5c4 5.10.47-linuxkit #1 SMP Sat Jul 3 21:51:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

2. I think so but here's more info about what I did:

     1. First I opened the data science dev container from vscode-dev-containers
     2. Then I ran `python -m venv mytestEnv`
     3. Then I opened the `mytestEnv ` folder (still inside that container)

Let me know if I should try something different instead

**Comment by rchiodo** [comment]
When you say this:

> Then I opened the mytestEnv folder (still inside that container)

Do you mean you opened this as a folder in VS code?

That would prevent that virtual environment from being listed. You should open the folder just above that.

**Comment by rchiodo** [comment]
And because of this bug here:
https://github.com/microsoft/vscode-jupyter/issues/7583 

you have to create the virtual env before opening the folder in VS code

**Comment by mjbvz** [comment]
Ah yes that was the issue! After opening the parent folder, I could select the kernel and then install worked properly

**Comment by rchiodo** [comment]
I'm going to open another issue to make us install pip. Seems weird that we can't just install it.

**Comment by DonJayamanne** [comment]
> I'm going to open another issue to make us install pip. Seems weird that we can't just install it.

We're waiting on telemetry for this, Brett didn't want to install Pip as he didn't believe it was the right solution when I submitted a PR to do that in the Python extension.

@rchiodo /cc
